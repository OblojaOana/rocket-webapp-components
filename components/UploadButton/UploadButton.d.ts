export interface UploadButtonProps {

    /**
     * The text displayed when user hover over the button
     */
    title?: string;

    /**
     * Callback fired when a "click" event is detected
     * @param {object} event The event source of the callback.
     */
    onClick: (event: React.SyntheticEvent) => void

    /**
     * If set to true, the button will be disabled
     */
    disabled?: boolean

    /**
     * The content of the button
     */
    children?: React.ReactNode;

    /**
     * @default "theme" 
    * The color of the button
    
     */
    color?: "primary" |
            "info" |
            "theme" |
            "themeNoBackground" |
            "themeWithBackground" |
            "success" |
            "warning" |
            "danger" |
            "rose" |
            "white" |
            "simple" |
            "defaultNoBackground" |
            "primaryNoBackground" |
            "infoNoBackground" |
            "successNoBackground" |
            "warningNoBackground" |
            "dangerNoBackground" |
            "roseNoBackground"

    /**
     * @default "medium" 
    * The size of the button
    
     */
    size?: "small" |
           "medium"

    /**
     * @default "small"
     * The size of the font
     
     */
    fontSize?: "inherit" |
                "default" |
                "small" |
                "large"

    /**
    * Demos:
    * 
    * - https://bit.dev/totalsoft_oss/react-mui/upload-button
    * 
    *  Take a look over the code for further information:https://github.com/osstotalsoft/rocket-webapp-components/blob/develop/components/UploadButton/UploadButton.js
    */

}

export default function UploadButton(props: UploadButtonProps): JSX.Element;