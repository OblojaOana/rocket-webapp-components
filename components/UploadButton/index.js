import UploadButton from "../../core/UploadButton";
export default UploadButton;