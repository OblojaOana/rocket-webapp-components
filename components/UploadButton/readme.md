##### Copy this snapshot in the overview panel

```js
import React from "react";
import UploadButton from "@bit/totalsoft_oss.react-mui.upload-button";

export default <UploadButton title="Upload"></UploadButton>;
```
