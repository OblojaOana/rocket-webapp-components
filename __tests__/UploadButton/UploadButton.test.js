import React from "react";
import renderer from "react-test-renderer";
import UploadButton from "../../components/UploadButton/UploadButton";

describe("Render UploadButton", () => {
  it("render correctly UploadButton component", () => {
    const component = renderer.create(<UploadButton title="Upload" />);

    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
