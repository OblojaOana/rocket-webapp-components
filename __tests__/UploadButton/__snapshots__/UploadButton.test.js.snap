// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadButton Snapshot 1`] = `
<button
  aria-describedby={null}
  aria-label="Upload"
  className="MuiButtonBase-root MuiIconButton-root makeStyles-button-1 makeStyles-theme-17"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  title="Upload"
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM14 13v4h-4v-4H7l5-5 5 5h-3z"
      />
    </svg>
  </span>
</button>
`;
